hydra-main/src/main/java/com/addthis/hydra/job/spawn/balancer/JobTaskItem.java

```diff
@@ -33,7 +33,7 @@ public JobTask getTask() {
 
     @Override
     public boolean equals(Object o) {
-        if (o.getClass() != getClass()) {
+        if (o == null || o.getClass() != getClass()) {
```

Comment:
A fun alternative here is a humble instanceof call. It'll return false for both nulls and guys two lines away from a class cast exception.

One time I checked the jvm bytecode specifically to spite an old professor and at the time at least, it was also several instructions more efficient.